### DISCUSSION TITLE
Welcome to pylinac Discussions!

### DISCUSSION THREAD

**Discussion by jrkerns** [jrkerns/pylinac#351] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by kegge13** [comment]
My name is Theo and I'm the author of BistroMath (https://github.com/kegge13/BistroMath). It contains a lot of thoughts on how to analyse profiles. As member of the NCS-33 subcommittee I can promote these concepts more widely.
I made a handy page as introduction to the concepts: https://bistromath.kegge13.nl/index.php?i=manual/hs29.htm. Open for discussion!

**Comment by LuisOlivaresJ** [comment]
Hi,
My name is Luis Olivares, and I am a medical physicist in radiotherapy. I am from Mexico.

I am very interested in use pylinac for some tasks in my hospital, so I am studying the API documentation. 
Also, in my free times I am working in a project for film dosimetry and its use for patient-specific quality assurance verifications (you can see it in [GitHub](https://dosepy.readthedocs.io/en/latest/intro.html) although it is still immature). In the future, I am planning to use some pylinac class in my project, at least pylinac.core.image.

I would like to contribute to pylinac because a believe it is an excellent way I can learn and build experience in it. If you have some tasks please let me kwon. I will be very happy to be a pylinac contributor.

Luis

**Reply by jrkerns** [reply]
Hello Luis, 
     Thanks for posting and welcome. You are free to use pylinac in your projects and also to fork it for your own needs. 
     
Your project looks pretty cool! I have always stayed away from film because I never needed it and DICOM was much more straightforward, so it's good to see someone coming up with an open-source solution for it! I encourage you to keep going. Your documentation is a great start. I feel that clear documentation was a big contributor to pylinac's popularity. 

I'm happy you want to contribute. I've never laid out a vision plan or roadmap for pylinac. Usually, the features are requests from customers. Also, most users of pylinac aren't software developers like other libraries might be, so the number of contributions is understandably low. I'm also kind of particular and since I want to be responsible for the users, I've never tried to solicit the community for contributions. 

All this to say, if you want to write something like a new method or see an issue (as you already have done and fixed) then I'm open to it. If you want to write a new algorithm then that's something to chat about first. You should also check out the [pymedphys](https://github.com/manorina/pymedphys) project as they are made up of a group of diverse contributors, unlike pylinac 😄. My goal was never to create something popular, but rather something I could use myself in the clinic. It just so happens it's open-source haha! 

Another approach I have recommended to others is to develop whatever thing you want in your own fork or in your own library and then if you think it would be a good fit for pylinac, you can link it and we can talk about making a PR for it. That way you get the functionality you need when you need it and we can later see about generalizing it. 

Keep up the good work!

**Comment by Daranporn** [comment]
what's 'self' in pylinac code?

**Reply by jrkerns** [reply]
From ChatGPT: "In object-oriented programming, classes are blueprints for creating objects. Objects are instances of a class, and they have their own set of attributes (data) and methods (functions) that can operate on that data. When you define a method within a class, you need a way to refer to the specific instance of the class that the method is being called on. This is where the `self` parameter comes into play.

When you define a method within a class, you include `self` as the first parameter in the method's parameter list. This parameter serves as a reference to the instance of the class. By convention, it is named `self`, although you can technically name it anything you want (but using `self` is a widely accepted practice).

Here's a simple example to illustrate this concept:

```python
class MyClass:
    def __init__(self, value):
        self.value = value
    
    def print_value(self):
        print(self.value)

# Creating an instance of MyClass
obj = MyClass(42)

# Calling the method using the instance
obj.print_value()
```

In this example, the `self` parameter in the `__init__` and `print_value` methods refers to the instance of `MyClass` that is created with `obj = MyClass(42)`. Inside the methods, you can access instance variables (like `self.value`) and call other methods on `self`.

In the context of your work at Radformation, you might use `self` extensively within Django models and views. For instance, when defining methods in a Django model, you'd use `self` to access and modify the attributes of the instance being worked on. This is crucial for maintaining data integrity and encapsulation within your application."

**Reply by LuisOlivaresJ** [reply]
In few words, ```self``` is a reference to the instance you are working with. It is the syntax used by python to have access to the attributes and methods of the self instance.

**Comment by Daranporn** [comment]
Can I generate a fluence map at each control point? If so, what is the code?
Now, I can generate a fluence map from each subbeam using this code.

`tlog.subbeams[1].fluence.actual.plot_map()`

Thank you so much

**Reply by jrkerns** [reply]
No, not out of the box. You'd have to generate it yourself.

**Reply by Daranporn** [reply]
Thank you. 👍
